src/ccl_background.c

```diff
@@ -833,98 +811,57 @@ double ccl_comoving_angular_distance(ccl_cosmology * cosmo, double a, int* statu
       ccl_check_status(cosmo, status);
     }
-    
+
     double chi;
     int gslstatus = gsl_spline_eval_e(cosmo->data.chi, a,
                                       cosmo->data.accelerator,&chi);
     if(gslstatus != GSL_SUCCESS) {
       ccl_raise_gsl_warning(gslstatus, "ccl_background.c: ccl_comoving_angular_distance():");
       *status |= gslstatus;
       ccl_cosmology_set_status_message(cosmo, "ccl_background.c: ccl_comoving_angular_distance(): Scale factor outside interpolation range.\n");
-      return NAN;      
+      return NAN;
     }
     return ccl_sinn(cosmo,chi,status);
   }
 }
 
-void ccl_comoving_angular_distances(ccl_cosmology * cosmo, int na, double a[], 
+void ccl_comoving_angular_distances(ccl_cosmology * cosmo, int na, double a[],
                                     double output[], int* status)
 {
-  if (!cosmo->computed_distances) {
-    ccl_cosmology_compute_distances(cosmo, status);
-    ccl_check_status(cosmo, status);
-  }
-  double chi;
-  int gslstatus;
+  int _status;
+
   for (int i=0; i < na; i++) {
-    if((a[i] > (1. - 1.e-8)) && (a[i]<=1.))
-      output[i]=0.;
-    else if(a[i]>1.) {
-      *status = CCL_ERROR_COMPUTECHI;
-      ccl_cosmology_set_status_message(cosmo, "ccl_background.c: scale factor cannot be larger than 1.\n");
-      ccl_check_status(cosmo,status);
-    }
-    else {
-      gslstatus = gsl_spline_eval_e(cosmo->data.chi,a[i],cosmo->data.accelerator,&chi);
-      output[i] = ccl_sinn(cosmo,chi,status);
-      if(gslstatus != GSL_SUCCESS) {
-        ccl_raise_gsl_warning(gslstatus, "ccl_background.c: ccl_comoving_angular_distances():");
-        *status |= gslstatus;
-        ccl_cosmology_set_status_message(cosmo, "ccl_background.c: ccl_comoving_angular_distances(): Scale factor outside interpolation range.\n");
-        output[i] =  NAN;      
-      }
-    }
+    _status = 0;
+    output[i] = ccl_comoving_angular_distance(cosmo, a[i], &_status);
+    *status |= _status;
   }
 }
 
 double ccl_luminosity_distance(ccl_cosmology * cosmo, double a, int* status)
 {
-  if((a > (1.0 - 1.e-8)) && (a<=1.0)) {
-    return 0.;
-  }
-  else if(a>1.) {
-    *status = CCL_ERROR_COMPUTECHI;
-    ccl_cosmology_set_status_message(cosmo, "ccl_background.c: scale factor cannot be larger than 1.\n");
-    ccl_check_status(cosmo,status);
-    return 0.;
-  }
-  else {
-    if (!cosmo->computed_distances) {
-      ccl_cosmology_compute_distances(cosmo, status);
-      ccl_check_status(cosmo, status);
-    }
-    return ccl_comoving_angular_distance(cosmo, a, status) / a;
-  }
+  return ccl_comoving_angular_distance(cosmo, a, status) / a;
 }
 
 void ccl_luminosity_distances(ccl_cosmology * cosmo, int na, double a[], double output[], int * status)
 {
-  if (!cosmo->computed_distances) {
-    ccl_cosmology_compute_distances(cosmo, status);
-    ccl_check_status(cosmo, status);
-  }
+  int _status;
+
   for (int i=0; i<na; i++) {
-    if ((a[i] > (1. - 1.e-8)) && (a[i] <= 1.)) 
```

Comment:
So this check only happens in comoving_angular_distance now and all other distance functions call that one?

Reply:
that is right.